### PR TITLE
Improve fuzzy search in clickhouse-client/clickhouse-local

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Improve fuzzy search in clickhouse-client/clickhouse-local (and fix repaiting)

I found few difference in sk/fzf and after this patch I think that it
should behave better.

So the following commands will be used by default:
- `fzf --tac --tiebreak=index`
- `sk --tac --tiebreak=-score`

Those two more or less allows true fuzzy reserve search.

### CODE REVIEW

File: base/base/ReplxxLineReader.h [hunk 1]
```diff
@@ -4,6 +4,12 @@
 
 #include <replxx.hxx>
 
+enum FuzzyFinderType
+{
+    FUZZY_FINDER_NONE,
+    FUZZY_FINDER_FZF,
```
Comment (alexey-milovidov) [inline base/base/ReplxxLineReader.h:10]: `fzf` and `skim` are rather new software projects (not universally known), let's at least add links to their repositories in a comment.

File: base/base/ReplxxLineReader.cpp [hunk 2]
```diff
@@ -515,9 +517,20 @@ void ReplxxLineReader::openInteractiveHistorySearch()
...
+            fuzzy_finder_command += " --tac --tiebreak=-score";
+            break;
+        case FUZZY_FINDER_FZF:
+            fuzzy_finder_command += " --tac --tiebreak=index";
+            break;
+        case FUZZY_FINDER_NONE:
+            /// assertion for !fuzzy_finder.empty() is enough
+            break;
+    }
+    fuzzy_finder_command += fmt::format(" < {} > {}", history_file.getPath(), output_file.getPath());
```
[the remark this answers is not in the document]
Comment (azat) [reply base/base/ReplxxLineReader.cpp]: Though I'm planning to rewrite this part, but let's add the escaping for now. Done.

**Comment by azat** [comment]
Tests unrelated:

>AST fuzzer (ubsan) — ../src/Common/assert_cast.h:50:12: runtime error: downcast of address 0x7f3f36f34d28 which does not point to an object of type 'const DB::AS

- #43907

>https://s3.amazonaws.com/clickhouse-test-reports/43498/7a9adbc2859c2469a575fff7817ae432289c68c8/stateful_tests__tsan_.html

- `00176_bson_parallel_parsing` - #43696 

>Stress test (asan) — Backward compatibility check: Error message in clickhouse-server.log (see bc_check_error_messages.txt)

```
2022.11.23 21:21:20.633856 [ 305365 ] {} <Error> test_chzowvu0.src_6: while loading part 4_15_15_0 on path data/test_chzowvu0/src_6/4_15_15_0: Code: 499. DB::Exception: The specified key does not exist.: while reading key: test/krb/gvyhoppphlyyrpiunzqccwbprfwhl, from bucket: test. (S3_ERROR), Stack trace (when copying this message, always include the lines below):
```

>Stress test (debug) — Backward compatibility check: Error message in clickhouse-server.log (see bc_check_error_messages.txt)

```
2022.11.23 09:38:21.613691 [ 260964 ] {} <Error> virtual void DB::MetadataStorageFromDiskTransaction::commit(): Code: 566. DB::ErrnoException: Cannot rmdir /var/lib/clickhouse/disks/s3/data/system/part_log/delete_tmp_202211_371_371_0, errno: 2, strerror: No such file or directory. (CANNOT_RMDIR), Stack trace (when copying this message, always include the lines below):
```

>Stress test (ubsan) — Backward compatibility check: Error message in clickhouse-server.log (see bc_check_error_messages.txt)

```
2022.11.23 09:10:16.315524 [ 367504 ] {} <Error> DiskObjectStorageTransaction: An error occurred while executing transaction's operation #0 (RemoveManyObjectStorageOperation (paths size: 11, keep all batch false, files to keep )): Code: 107. DB::ErrnoException: Cannot open file /var/lib/clickhouse/disks/s3/data/system/part_log/delete_tmp_202211_1_281_180/data.mrk3, errno: 2, strerror: No such file or directory. (FILE_DOESNT_EXIST), Stack trace (when copying this message, always include the lines below):
```

>Stress test (msan) — Backward compatibility check: OOM killer (or signal 9) in clickhouse-server.log

Looks like slow parts removal, **but for the previous version**

<details>

```
2022-11-23 09:25:22 Thread 75 (Thread 0x7efc3b0d0700 (LWP 233614)):
2022-11-23 09:25:22 #0  0x00007efdc484b99f in poll () from /lib/x86_64-linux-gnu/libc.so.6
2022-11-23 09:25:22 #1  0x000000000c09cd11 in poll ()
2022-11-23 09:25:22 #2  0x000000005520c715 in Poco::Net::SocketImpl::pollImpl (this=this@entry=0x703002aa75e0, remainingTime=..., mode=mode@entry=1) at ../contrib/poco/Net/src/SocketImpl.cpp:471
2022-11-23 09:25:22 #3  0x000000005520ce01 in Poco::Net::SocketImpl::poll (this=0x703002aa75e0, timeout=..., mode=1) at ../contrib/poco/Net/src/SocketImpl.cpp:540
2022-11-23 09:25:22 #4  0x000000005520a459 in Poco::Net::SocketImpl::receiveBytes (this=0x703002aa75e0, buffer=0x720000451000, length=4096, flags=0) at ../contrib/poco/Net/src/SocketImpl.cpp:344
2022-11-23 09:25:22 #5  0x00000000552198fc in Poco::Net::StreamSocket::receiveBytes (this=<optimized out>, buffer=<optimized out>, length=<optimized out>, flags=0) at ../contrib/poco/Net/src/StreamSocket.cpp:130
2022-11-23 09:25:22 #6  0x00000000551960bd in Poco::Net::HTTPSession::receive (buffer=0x720000451000 "ult><?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<DeleteResult xmlns=\"http://s3.amazonaws.com/doc/2006-03-01/\"><Deleted><Key>test/cvd/lbbyqepbdmihtmoaihudgqtntzbeb</Key></Deleted></DeleteResult>1/s3/aws4_re"..., length=4096, this=<optimized out>) at ../contrib/poco/Net/src/HTTPSession.cpp:166
2022-11-23 09:25:22 #7  Poco::Net::HTTPSession::refill (this=<optimized out>) at ../contrib/poco/Net/src/HTTPSession.cpp:183
2022-11-23 09:25:22 #8  0x0000000055195e60 in Poco::Net::HTTPSession::get (this=0xfffffffffffffdfc) at ../contrib/poco/Net/src/HTTPSession.cpp:113
2022-11-23 09:25:22 #9  0x000000005517d787 in Poco::Net::HTTPHeaderStreamBuf::readFromDevice (this=<optimized out>, buffer=0x72000182c004 " /test?delete HTTP/1.1\r\namz-sdk-invocation-id: 88B3B652-73AE-4CC8-B9EE-7D46F61AFD7C\r\namz-sdk-request: attempt=1\r\nauthorization: AWS4-HMAC-SHA256 Credential=clickhouse/20221123/us-east-1/s3/aws4_reques"..., length=4092) at ../contrib/poco/Net/src/HTTPHeaderStream.cpp:49
2022-11-23 09:25:22 #10 0x000000005515c9f0 in Poco::BasicBufferedStreamBuf<char, std::__1::char_traits<char>, Poco::Net::HTTPBufferAllocator>::underflow (this=0x7efc3b0c46e8) at ../contrib/poco/Foundation/include/Poco/BufferedStreamBuf.h:97
2022-11-23 09:25:22 #11 0x0000000060f1cc88 in std::__1::basic_streambuf<char, std::__1::char_traits<char> >::uflow (this=0x7efc3b0c46e8) at ../contrib/libcxx/include/streambuf:445
2022-11-23 09:25:22 #12 0x0000000060f247ee in std::__1::basic_streambuf<char, std::__1::char_traits<char> >::sbumpc[abi:v15003]() (this=0xfffffffffffffdfc) at ../contrib/libcxx/include/streambuf:191
2022-11-23 09:25:22 #13 std::__1::basic_istream<char, std::__1::char_traits<char> >::get (this=0x7efc3b0c4750) at ../contrib/libcxx/include/istream:756
2022-11-23 09:25:22 #14 0x000000005518f39d in Poco::Net::HTTPResponse::read (this=0x7efc3b0c4980, istr=...) at ../contrib/poco/Net/src/HTTPResponse.cpp:235
2022-11-23 09:25:22 #15 0x0000000055166cae in Poco::Net::HTTPClientSession::receiveResponse (this=0x714001f74c18, response=...) at ../contrib/poco/Net/src/HTTPClientSession.cpp:315
2022-11-23 09:25:22 #16 0x000000003f47e84e in DB::S3::PocoHTTPClient::makeRequestInternal (this=this@entry=0x71200065c298, request=..., response=...) at ../src/IO/S3/PocoHTTPClient.cpp:381
2022-11-23 09:25:22 #17 0x000000003f478842 in DB::S3::PocoHTTPClient::MakeRequest (this=0x71200065c298, request=..., readLimiter=<optimized out>, writeLimiter=<optimized out>) at ../src/IO/S3/PocoHTTPClient.cpp:150
2022-11-23 09:25:22 #18 0x0000000055ad5d1a in Aws::Client::AWSClient::AttemptOneRequest(std::__1::shared_ptr<Aws::Http::HttpRequest> const&, Aws::AmazonWebServiceRequest const&, char const*, char const*, char const*) const ()
2022-11-23 09:25:22 #19 0x0000000055acd88c in Aws::Client::AWSClient::AttemptExhaustively(Aws::Http::URI const&, Aws::AmazonWebServiceRequest const&, Aws::Http::HttpMethod, char const*, char const*, char const*) const ()
2022-11-23 09:25:22 #20 0x0000000055af69fb in Aws::Client::AWSXMLClient::MakeRequest(Aws::Http::URI const&, Aws::AmazonWebServiceRequest const&, Aws::Http::HttpMethod, char const*, char const*, char const*) const ()
2022-11-23 09:25:22 #21 0x0000000055d1fedd in Aws::S3::S3Client::DeleteObjects(Aws::S3::Model::DeleteObjectsRequest const&) const ()
2022-11-23 09:25:22 #22 0x0000000041b559c5 in DB::S3ObjectStorage::removeObjectsImpl (this=<optimized out>, objects=..., if_exists=false) at ../src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp:398
2022-11-23 09:25:22 #23 0x0000000041b578f7 in DB::S3ObjectStorage::removeObjects (this=0x7efc3b0c41e8, objects=...) at ../src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp:419
2022-11-23 09:25:22 #24 0x0000000041a48129 in DB::(anonymous namespace)::RemoveObjectStorageOperation::finalize (this=<optimized out>) at ../src/Disks/ObjectStorages/DiskObjectStorageTransaction.cpp:137
2022-11-23 09:25:22 #25 0x0000000041a3edb7 in DB::DiskObjectStorageTransaction::commit (this=0x7070004d3818) at ../src/Disks/ObjectStorages/DiskObjectStorageTransaction.cpp:749
2022-11-23 09:25:22 #26 0x00000000419bd358 in DB::DiskObjectStorage::removeSharedFile (this=<optimized out>, path=..., delete_metadata_only=false) at ../src/Disks/ObjectStorages/DiskObjectStorage.cpp:259
2022-11-23 09:25:22 #27 0x00000000419d6b56 in DB::DiskObjectStorage::removeFile (this=0x7efc3b0c41e8, path=...) at ../src/Disks/ObjectStorages/DiskObjectStorage.h:81
2022-11-23 09:25:22 #28 0x0000000041aec15b in DB::DiskDecorator::removeFile (this=<optimized out>, path=...) at ../src/Disks/DiskDecorator.cpp:144
2022-11-23 09:25:22 #29 0x0000000041ae0aa8 in DB::DiskRestartProxy::removeFile (this=0x710000064b18, path=...) at ../src/Disks/DiskRestartProxy.cpp:235
2022-11-23 09:25:22 #30 0x0000000048ed9e3b in DB::MergeTreeMutationEntry::removeFile (this=<optimized out>) at ../src/Storages/MergeTree/MergeTreeMutationEntry.cpp:102
2022-11-23 09:25:22 #31 0x0000000048e604dc in DB::StorageMergeTree::clearOldMutations (this=this@entry=0x71e0014d9c20, truncate=<optimized out>) at ../src/Storages/StorageMergeTree.cpp:1290
2022-11-23 09:25:22 #32 0x0000000048ec292a in DB::StorageMergeTree::scheduleDataProcessingJob(DB::BackgroundJobsAssignee&)::$_4::operator()() const (this=<optimized out>) at ../src/Storages/StorageMergeTree.cpp:1215
2022-11-23 09:25:22 #33 std::__1::__invoke[abi:v15003]<DB::StorageMergeTree::scheduleDataProcessingJob(DB::BackgroundJobsAssignee&)::$_4&> (__f=...) at ../contrib/libcxx/include/__functional/invoke.h:394
2022-11-23 09:25:22 #34 std::__1::__invoke_void_return_wrapper<bool, false>::__call<DB::StorageMergeTree::scheduleDataProcessingJob(DB::BackgroundJobsAssignee&)::$_4&>(DB::StorageMergeTree::scheduleDataProcessingJob(DB::BackgroundJobsAssignee&)::$_4&) (__args=...) at ../contrib/libcxx/include/__functional/invoke.h:470
2022-11-23 09:25:22 #35 std::__1::__function::__default_alloc_func<DB::StorageMergeTree::scheduleDataProcessingJob(DB::BackgroundJobsAssignee&)::$_4, bool ()>::operator()[abi:v15003]() (this=<optimized out>) at ../contrib/libcxx/include/__functional/function.h:235
2022-11-23 09:25:22 #36 std::__1::__function::__policy_invoker<bool ()>::__call_impl<std::__1::__function::__default_alloc_func<DB::StorageMergeTree::scheduleDataProcessingJob(DB::BackgroundJobsAssignee&)::$_4, bool ()> >(std::__1::__function::__policy_storage const*) (__buf=<optimized out>) at ../contrib/libcxx/include/__functional/function.h:716
2022-11-23 09:25:22 #37 0x0000000047a17217 in std::__1::__function::__policy_func<bool ()>::operator()[abi:v15003]() const (this=0x7efc3b0c41e8) at ../contrib/libcxx/include/__functional/function.h:848
2022-11-23 09:25:22 #38 std::__1::function<bool ()>::operator()() const (this=0x7efc3b0c41e8) at ../contrib/libcxx/include/__functional/function.h:1197
2022-11-23 09:25:22 #39 DB::ExecutableLambdaAdapter::executeStep (this=0x70b000d555b8) at ../src/Storages/MergeTree/IExecutableTask.h:59
2022-11-23 09:25:22 #40 0x00000000483d1818 in DB::MergeTreeBackgroundExecutor<DB::OrdinaryRuntimeQueue>::routine (this=<optimized out>, item=...) at ../src/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp:137
2022-11-23 09:25:22 #41 0x00000000483d5671 in DB::MergeTreeBackgroundExecutor<DB::OrdinaryRuntimeQueue>::threadFunction (this=0x714000278e18) at ../src/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp:259
2022-11-23 09:25:22 #42 0x00000000483e2589 in DB::MergeTreeBackgroundExecutor<DB::OrdinaryRuntimeQueue>::MergeTreeBackgroundExecutor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned long, unsigned long, unsigned long)::{lambda()#1}::operator()() const (this=<optimized out>) at ../src/Storages/MergeTree/MergeTreeBackgroundExecutor.h:185
2022-11-23 09:25:22 #43 std::__1::__invoke[abi:v15003]<DB::MergeTreeBackgroundExecutor<DB::OrdinaryRuntimeQueue>::MergeTreeBackgroundExecutor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned long, unsigned long, unsigned long)::{lambda()#1}&> (__f=...) at ../contrib/libcxx/include/__functional/invoke.h:394
2022-11-23 09:25:22 #44 std::__1::__invoke_void_return_wrapper<void, true>::__call<DB::MergeTreeBackgroundExecutor<DB::OrdinaryRuntimeQueue>::MergeTreeBackgroundExecutor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned long, unsigned long, unsigned long)::{lambda()#1}&>(DB::MergeTreeBackgroundExecutor<DB::OrdinaryRuntimeQueue>::MergeTreeBackgroundExecutor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned long, unsigned long, unsigned long)::{lambda()#1}&) (__args=...) at ../contrib/libcxx/include/__functional/invoke.h:479
2022-11-23 09:25:22 #45 std::__1::__function::__default_alloc_func<DB::MergeTreeBackgroundExecutor<DB::OrdinaryRuntimeQueue>::MergeTreeBackgroundExecutor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned long, unsigned long, unsigned long)::{lambda()#1}, void ()>::operator()[abi:v15003]() (this=<optimized out>) at ../contrib/libcxx/include/__functional/function.h:235
2022-11-23 09:25:22 #46 std::__1::__function::__policy_invoker<void ()>::__call_impl<std::__1::__function::__default_alloc_func<DB::MergeTreeBackgroundExecutor<DB::OrdinaryRuntimeQueue>::MergeTreeBackgroundExecutor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned long, unsigned long, unsigned long)::{lambda()#1}, void ()> >(std::__1::__function::__policy_storage const*) (__buf=<optimized out>) at ../contrib/libcxx/include/__functional/function.h:716
2022-11-23 09:25:22 #47 0x000000002897bbe5 in std::__1::__function::__policy_func<void ()>::operator()[abi:v15003]() const (this=0x7efc3b0c7270) at ../contrib/libcxx/include/__functional/function.h:848
2022-11-23 09:25:22 #48 std::__1::function<void ()>::operator()() const (this=0x7efc3b0c7270) at ../contrib/libcxx/include/__functional/function.h:1197
2022-11-23 09:25:22 #49 ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::worker (this=this@entry=0x714000278ef8, thread_it=thread_it@entry=...) at ../src/Common/ThreadPool.cpp:294
2022-11-23 09:25:22 #50 0x0000000028988c89 in ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}::operator()() const (this=<optimized out>) at ../src/Common/ThreadPool.cpp:144
2022-11-23 09:25:22 #51 std::__1::__invoke[abi:v15003]<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}&> (__f=...) at ../contrib/libcxx/include/__functional/invoke.h:394
2022-11-23 09:25:22 #52 std::__1::__apply_tuple_impl[abi:v15003]<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}&, std::__1::tuple<>&>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}&, std::__1::tuple<>&, std::__1::__tuple_indices<>) (__f=..., __t=...) at ../contrib/libcxx/include/tuple:1789
2022-11-23 09:25:22 #53 std::__1::apply[abi:v15003]<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}&, std::__1::tuple<>&>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}&, std::__1::tuple<>&) (__f=..., __t=...) at ../contrib/libcxx/include/tuple:1798
2022-11-23 09:25:22 #54 ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}&&)::{lambda()#1}::operator()() (this=<optimized out>) at ../src/Common/ThreadPool.h:196
2022-11-23 09:25:22 #55 0x0000000028988b09 in std::__1::__invoke[abi:v15003]<ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}&&)::{lambda()#1}&> (__f=...) at ../contrib/libcxx/include/__functional/invoke.h:394
2022-11-23 09:25:22 #56 std::__1::__invoke_void_return_wrapper<void, true>::__call<ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}&&)::{lambda()#1}&>(ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}&&)::{lambda()#1}&) (__args=...) at ../contrib/libcxx/include/__functional/invoke.h:479
2022-11-23 09:25:22 #57 std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}&&)::{lambda()#1}, void ()>::operator()[abi:v15003]() (this=0xfffffffffffffdfc) at ../contrib/libcxx/include/__functional/function.h:235
2022-11-23 09:25:22 #58 std::__1::__function::__policy_invoker<void ()>::__call_impl<std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false> >::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}&&)::{lambda()#1}, void ()> >(std::__1::__function::__policy_storage const*) (__buf=<optimized out>) at ../contrib/libcxx/include/__functional/function.h:716
2022-11-23 09:25:22 #59 0x000000002897302d in std::__1::__function::__policy_func<void ()>::operator()[abi:v15003]() const (this=0x7efc3b0c77e0) at ../contrib/libcxx/include/__functional/function.h:848
2022-11-23 09:25:22 #60 std::__1::function<void ()>::operator()() const (this=0x7efc3b0c77e0) at ../contrib/libcxx/include/__functional/function.h:1197
2022-11-23 09:25:22 #61 ThreadPoolImpl<std::__1::thread>::worker (this=0x70f0000069f0, thread_it=...) at ../src/Common/ThreadPool.cpp:294
2022-11-23 09:25:22 #62 0x0000000028981063 in ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}::operator()() const (this=<optimized out>) at ../src/Common/ThreadPool.cpp:144
2022-11-23 09:25:22 #63 std::__1::__invoke[abi:v15003]<ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}> (__f=...) at ../contrib/libcxx/include/__functional/invoke.h:394
2022-11-23 09:25:22 #64 std::__1::__thread_execute[abi:v15003]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}>&, std::__1::__tuple_indices<>) (__t=...) at ../contrib/libcxx/include/thread:284
2022-11-23 09:25:22 #65 std::__1::__thread_proxy[abi:v15003]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}> >(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, long, std::__1::optional<unsigned long>, bool)::{lambda()#2}>) (__vp=0x7020000bd480) at ../contrib/libcxx/include/thread:295
2022-11-23 09:25:22 #66 0x00007efdc4933609 in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
2022-11-23 09:25:22 #67 0x00007efdc4858133 in clone () from /lib/x86_64-linux-gnu/libc.so.6
```

</details>

**Comment by antonio2368** [comment]
> I found few difference in sk/fzf and after this patch I think that it
should behave better.

@azat Can you provide more information about this? I'm honestly surprised that they have different behaviour for same `tiebreak` mode.

**Comment by azat** [comment]
>Can you provide more information about this?

- `fzf` don't hvae `tiebreak=score`
- `fzf` cannot handle `-` for criteria
- `sk --tiebreak=[-]index` has "fuzzy match" prefered over the position, while for the queries (and actually to the bash history in my opinion) the real fuzzying is not that important.

About `sk` you can see the difference using this examples:
- `yes foobar | head -n1000 | cat -n | sk --tac --tiebreak=-score`
- `yes foobar | head -n1000 | cat -n | sk --tac --tiebreak=index`
Search for `1000`

**Review by antonio2368 — APPROVED** [review]

**Pull request merged**